### PR TITLE
[CENNSO-148] Update to VPP 22.10

### PR DESCRIPTION
**Pull request by ivan4th**
This PR needs to be updated after travelping/fpp-vpp#40 is merged (bump `vpp.spec`)

### CODE REVIEW

File: upf/upf_pfcp.c [hunk 1]
```diff
@@ -2575,10 +2575,9 @@ process_urrs (vlib_main_t * vm, upf_session_t * sess,
 
   if (PREDICT_FALSE (status != URR_OK))
     {
-      vec_validate_ha (uev, 0, sizeof (upf_event_urr_hdr_t), 0);
-      ueh =
-	(upf_event_urr_hdr_t *) vec_header (uev,
-					    sizeof (upf_event_urr_hdr_t));
+      /* TBD: re-verify! */
```
Comment (mogaika) [inline upf/upf_pfcp.c:2578]: `vec_validate_hap` change looks valid, header size need to be provided in case of empty vector. What to verify?

**Review by mogaika — APPROVED** [review]
Only minor notes

File: upf/pfcp.c [hunk 2]
```diff
@@ -8102,16 +8105,16 @@ encode_ie (const struct pfcp_group_ie_def *item,
...
+    _vec_find (*vec)->len = hdr;
+
+/* #if CLIB_DEBUG > 0 */
+/*   /\* */
+/*    * Make sure that we didn't have heap corruption */
+/*    *\/ */
+/*   ASSERT (_vec_len (*vec) <= */
+/* 	  clib_mem_size ((void *) (*vec) - vec_header_bytes (0)) - */
+/* 	  vec_header_bytes (0)); */
+/* #endif */
```
Comment (demo-exe) [inline upf/pfcp.c:8117]: do we leave this commented or behind some feature flag?

File: upf/upf_pfcp_server.c [hunk 3]
```diff
@@ -188,9 +188,9 @@ upf_pfcp_send_data (pfcp_msg_t * msg)
   at.lcl_port = msg->lcl.port;
   at.rmt_port = msg->rmt.port;
 
-  app_send_dgram_raw (s->tx_fifo, &at, mq, msg->data,
-		      _vec_len (msg->data), SESSION_IO_EVT_TX,
-		      1 /* do_evt */ , 0);
+  int r = app_send_dgram_raw (s->tx_fifo, &at, mq, msg->data,
```
Comment (demo-exe) [inline upf/upf_pfcp_server.c:191]: is this for some compiler unused diagnostic?

File: test/e2e/network/netns.go [hunk 4]
```diff
@@ -259,7 +226,7 @@ func (netns *NetNS) ListenUDP(ctx context.Context, laddr *net.UDPAddr) (*net.UDP
 	err := netns.Do(func() error {
 		var innerErr error
 		network := "udp4"
-		if netns.ipv6 {
+		if netns.ipv6 || laddr.IP.To4() == nil {
```
Comment (demo-exe) [inline test/e2e/network/netns.go:229]: is this a logic change ?

**Review by demo-exe — APPROVED** [review]

**Pull request merged**